### PR TITLE
Fix clashing contexts

### PR DESCRIPTION
**Pull request by DominicOram**
See also the test in https://github.com/bluesky/ophyd-async/pull/600

### CODE REVIEW

**Comment by AlexanderWells-diamond** [comment]
The CI failures here will be fixed by the changes that I'm working on in #50

File: tests/test_aioca.py
Comment (coretl) [inline tests/test_aioca.py:1]: This is the bug report from ophyd-async without interaction from pyepics: https://github.com/bluesky/ophyd-async/issues/135

This seems to:
- Do a caget successfully in the main thread
- Try to do caget in a different thread on the same PV and see it timeout

Maybe doing that in this test might show the bug?

Comment (DominicOram) [reply tests/test_aioca.py:1]: Great, that worked, the new test fails on main but passes here.

**Comment by DominicOram** [comment]
Are you happy to merge despite the CI failures @coretl?

**Comment by coretl** [comment]
@AlexanderWells-diamond how close are your CI updates? We can either merge and release this as is (possibly disabling tests to get it to publish), or wait for your updates?

**Comment by AlexanderWells-diamond** [comment]
I'm working on a full adoption of the Copier template, which will involve things like moving the source files around. See #52 which replaced #50.

**Comment by DominicOram** [comment]
Discussed with @coretl, he's happy to merge without full coverage

**Pull request merged**